### ISSUE TITLE
DotvvmRequestContext refactoring

### ISSUE THREAD

**Issue by tomasherceg** [riganti/dotvvm#406]
The `DotvvmRequestContext` object and `IDotvvmRequestContext` interfaces are too heavy and contain some internal stuff. 
We should discuss which members should be moved somewhere else. 

<table>
    <tr>
        <th>Member</th>
        <th>Verdict</th>
    </tr>
	<tr>
		<td><code>HttpContext</code></td>
		<td>keep</td>
	</tr>	
	<tr>
		<td><code>Configuration</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>ViewModel</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>Presenter</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>Route</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>IsPostBack</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>Parameters</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>ResourceManager</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>ModelState</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>Query</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>IsSpaRequest</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>IsInPartialRenderingMode</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>Services</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>InterruptRequest()</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>RedirectToUrl(...)</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>RedirectToRoute(...)</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>RedirectToUrlPermanent(...)</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>RedirectToRoutePermanent(...)</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>FailOnInvalidModelState()</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>TranslateVirtualPath(...)</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>ReturnFile(...)</code></td>
		<td>keep</td>
	</tr>
	<tr>
		<td><code>IsCommandExceptionHandled</code></td>
		<td>can be moved (breaking change)</td>
	</tr>
	<tr>
		<td><code>IsPageExceptionHandled</code></td>
		<td>can be moved (breaking change)</td>
	</tr>
	<tr>
		<td><code>CommandException</code></td>
		<td>can be moved (breaking change)</td>
	</tr>
	<tr>
		<td><code>ResultIdFragment</code></td>
		<td>is it working? undocumented</td>
	</tr>
	<tr>
		<td><code>ChangeCurrentCulture(...)</code></td>
		<td>will be obsolete soon</td>
	</tr>
	<tr>
		<td><code>GetCurrentUICulture()</code></td>
		<td>will be obsolete soon</td>
	</tr>
	<tr>
		<td><code>GetCurrentCulture()</code></td>
		<td>will be obsolete soon</td>
	</tr>
	<tr>
		<td><code>RequestTracers</code></td>
		<td>internal</td>
	</tr>
	<tr>
		<td><code>CsrfToken</code></td>
		<td>internal</td>
	</tr>
	<tr>
		<td><code>ReceivedViewModelJson</code></td>
		<td>internal</td>
	</tr>
	<tr>
		<td><code>GetSpaContentPlaceHolderUniqueId()</code></td>
		<td>internal</td>
	</tr>
	<tr>
		<td><code>ViewModelJson</code></td>
		<td>internal</td>
	</tr>
	<tr>
		<td><code>PostBackUpdatedControls</code></td>
		<td>internal</td>
	</tr>
	<tr>
		<td><code>GetSerializedViewModel()</code></td>
		<td>internal</td>
	</tr>
</table>

**Comment by exyi** [comment]
I tried to address this in 078092050ed620b224efb5f1abb7362779047960, sorry for pushing it directly and not opening a PR.

I have moved all methods out of the interface to a static class as extension methods. `RequestTracers` collection was moved to Scoped service, as there is hopefully no user of it. `PostBackUpdatedControls` was removed. Other properties are the same.

By the way yes, `ResultIdFragment` is working and there is a UI test for it.
